packages/wallets/src/components/WalletTourGuide/WalletTourGuide.tsx

```diff
@@ -1,31 +1,75 @@
-import React from 'react';
-import Joyride, { CallBackProps } from 'react-joyride';
+import React, { useEffect, useState } from 'react';
+import Joyride, { ACTIONS, CallBackProps } from 'react-joyride';
+import { useActiveWalletAccount, useAuthorize, useAvailableWallets, useWalletAccountsList } from '@deriv/api';
+import { WalletText } from '../Base';
 import { TooltipComponent, tourStepConfig } from './WalletTourGuideSettings';
 import './WalletTourGuide.scss';
 
-type TProps = {
-    isStarted?: boolean;
-    setIsStarted?: (value: boolean) => void;
-};
+const WalletTourGuide = () => {
+    const key = 'walletsOnboarding';
+    const [isStarted, setIsStarted] = useState(localStorage.getItem(key) === 'started');
+
+    const { switchAccount } = useAuthorize();
+    const { data: wallets } = useWalletAccountsList();
+    const { data: activeWallet } = useActiveWalletAccount();
+    const { data: availableWallets } = useAvailableWallets();
+
+    const fiatWalletLoginId = wallets?.[0]?.loginid;
+    const activeWalletLoginId = activeWallet?.loginid;
 
-const WalletTourGuide = ({ isStarted = false, setIsStarted }: TProps) => {
     const callbackHandle = (data: CallBackProps) => {
-        if (data.action === 'reset') {
-            setIsStarted?.(false);
+        const { action } = data;
+
+        if (action === ACTIONS.RESET) {
+            localStorage.removeItem(key);
+            setIsStarted(false);
         }
     };
 
+    useEffect(() => {
+        const switchToFiatWallet = (): boolean => {
+            if (fiatWalletLoginId && fiatWalletLoginId !== activeWalletLoginId) {
+                switchAccount(fiatWalletLoginId);
+                return true;
+            }
+            return false;
+        };
+
+        const onStorage = () => {
+            const needToStart = localStorage.getItem(key) === 'started';
+            if (needToStart) {
+                const isJustSwitched = switchToFiatWallet();
+                if (isJustSwitched) setTimeout(() => setIsStarted(true), 1000);
```

Comment:
whats the purpose of setTimeout here? what does it need to wait for?

Reply:
@wojciech-deriv it's because we need to switch to real fiat wallet (or to demo wallet if there is not real wallet) and start onboarding. Without this onboarding starts before wallet switching and it highlights wrong wallet header for 1st step

Reply:
what if switching is gonna take more than 1000ms? Typically its better to wait until we know for sure all the prerequisities are actually available. setTimeout tends to be really prone to race hazards (in so many crazy ways), and as a rule of thumb, should be strongly avoided (unless its valid use case for them)
It might require changing the state tree a bit - happy to jump on call if needed and pair on that.

Reply:
@wojciech-deriv if switching is more than 1000ms then 1st onboarding step will highlight wrong element :) I know what you mean, I've tried to use isSuccess and isFetching from useAuthorize hook and wait for this, but the main problem is we have setTimeout for 0.3 in WalletsAccordion when the user switches between wallets. I mean the user make API call for authorize with new token, we receive response from BE with new active loginid, in this case active wallet is new from now and it fires setTimeout for 0.3 sec to scroll to open wallet. Sometimes react-joyride works fine with it, but sometimes it stucks so I need this setTimeout at least for 0.3 sec.